[concepts-overview] Improve comprehension

bgilmer77 opened the pull request:
Signed-off-by: Brad Gilmer <brad@gilmer.tv>

Confectrician commented on concepts/index.md:
```diff
@@ -6,29 +6,38 @@ layout: documentation
-While the physical view is required for setup, configuration, troubleshooting, etc., the functional side covers the information that matter to applications, such as user interfaces and automation logic.
+openHAB is built around two distinctly different views.
+The first is the physical view - devices, configurations, connections, etc.
+The second is the function view.
```
functional

bgilmer77 replied:
Agree - this will be addressed in my next commit.

Confectrician commented on concepts/index.md:
```diff
@@ -6,29 +6,38 @@ layout: documentation
-**Things** are the entities that can be physically added to a system and which can potentially provide many functionalities at once. It is important to note that things do not have to be devices, but they can also represent a web service or any other manageable source of information and functionality.
-Things provide their functionality through a set of **Channels**. Channels are "passive" and can be regarded as a declaration of a Thing, what it can offer. It is up to the individual setup, which of the Channels are actively used through Items (see below).
+**Things** are the entities that can be physically added to a system.
+These Things may provide more than just one function (for example, a thermostat may allow you to set the room temperature and also turn on a fan).
```
Just to think about:
Maybe this example is not that easy to grasp.
(What has the fan to do with that thermostat now? How does this help me?)

But as i said just to think about. I have no better example as suggestoin now, but will think about what a better one could be.

bgilmer77 replied:
@Confectrician here in the U.S., thermostats have a "Fan on/off" button as well as a set of buttons to adjust the set point.  I know in Europe, many of the thermostats do not have the fan switch, so I agree this is not a good example.

How about this:
"Things may provide more than just one function (for example, a z-wave multi-sensor may provide a motion detector and also measure room temperature)"

I actually have one of those...

Confectrician replied:
Hey Brad,

Suggestion sounds good and I think it is easy to understand for everyone.

I am perfectly fine with it.
Good idea.

Confectrician commented on concepts/index.md:
```diff
@@ -6,29 +6,38 @@ layout: documentation
+
+The glue between Things and Items are **Links**.
+Links are associations between exactly one Thing Channel and one Item.
+If a Channel is linked to an Item, it is "enabled", which means that the functionality that the Item represents is accessible through the Channel.
```
Maybe its my non native english again, but i think there is one `that` more than needed.

bgilmer77 replied:
Agree.  Just so you know, it is acceptable gramatically as written, but I agree that taking the extra `that` out improves readability.  This will be addressed in my next commit.

Confectrician commented on concepts/index.md:
```diff
@@ -6,29 +6,38 @@ layout: documentation
+It has a physical address and it must be configured in openHAB in order to be used (remember the physical view introduced at the beginning of this article?).
 
+Functionally, the user is interested in controlling the two lights.
+In order for the user to achieve the desired functionality with the lights, the *Items* which represent the lights in openHAB are linked to the *Channels* of the actuator.
```
Consider changing this sentence to a single item explanation.

Something like:
"In order for the user to achieve the desired functionality with the lights, each light is linked as a Item to one of the actuators *Channels*."

bgilmer77 replied:
How about the following, which I am putting in my next commit?

The user wants to control the two lights.
In order to achieve this functionality, two *Items* representing the lights are linked to *Channels* of the actuator, the *Thing*.
In this case, one can think of a *Link* being equivalent to a physical wire.

bgilmer77 commented:
@Confectrician okay, I think this is ready for a review please.

ThomDietrich commented:
@bgilmer77 this article is only a replicate from: https://github.com/eclipse/smarthome/blob/master/docs/documentation/concepts/index.md
Please move your changes there.

I know that the situation is not idea. For addons we add a reminder regarding their origin to the article source. Maybe that should also be done for the concepts articles?

bgilmer77 commented:
@ThomDietrich okay - I will make that change and add a note.  How can I determine what pages are hosted in the OH github and which are derived from another source?  I would prefer not to have to go through this again in the future with another page.

bgilmer77 commented:
@ThomDietrich I have opened an Eclipse account, created issue #4659, and submitted a PR which addresses the issue.  Should I close this?

Confectrician commented:
I woul leave it open till the smarthome pr is merged.

Confectrician commented:
About external content:
https://github.com/openhab/openhab-docs#automatically-generated-parts
The list should be complete.

Confectrician commented:
Sorry for pushing again.
Just some references, to keep an overview of all related issues/prs:

eclipse/smarthome#4659
eclipse/smarthome#4660

ThomDietrich commented:
> https://github.com/openhab/openhab-docs#automatically-generated-parts
> The list should be complete.

Sadly it is not. There are single files which come from external sources. While I have some ideas how to deal with that issue in a better way, here I'll just quickly show you what's copied over:

In the file [pom.xml](https://github.com/openhab/openhab-docs/blob/gh-pages/pom.xml) search for "outputDirectory". You'll find that it so far covers the concepts folder, the `_data` folder all `_addons_*` folders, the article `installation/openhabian.md` and `configuration/habpanel.md`.

I was thinking about some bigger changes in that whole toolchain but am still wainting for serious progress with https://github.com/eclipse/smarthome/issues/4503 ... :-/

bgilmer77 commented:
The fact that some of the docs come from external sources leads to some strange results.  For example, I had modified the Concepts article to replace `Elcipse SmartHome` with `openHab`, but of course, removed those changes when I made the contribution in the ESH repo.  Not a big deal, but I suspect some OH documentation readers will wonder why this refers to ESH.

@ThomDietrich thanks for the info about `pom.xml`.  I will be sure to look at it in the future.

ThomDietrich commented:
Would you be able to remove mentioning of Eclipse SmartHome without loosing a clear focus?

bgilmer77 commented:
@ThomDietrich yes I can remove ESH.  I am not sure of the protocol here.  Being new, I don't want to barge into a new community and immediately remove references to their project in the documentation.

ThomDietrich commented:
Good point but you could just create the PR, then add a comment with the question if that step (out of reason xy) was correct.

bgilmer77 commented:
The PR is already created.  Are you suggesting to create a separate PR just for removal of the reference to ESH?  I can do that.  Or,
I could just make a commit with only that change, along with a comment and see what people have to say.  What is the best approach?

ThomDietrich commented:
Please always link issues and PRs on GitHub. Makes everything a bit easier. Here it is https://github.com/eclipse/smarthome/pull/4660

Now that you already have it there, just post the question as a comment and state that you'd like to clarify this before the merge is done.

Confectrician commented:
eclipse/smarthome#4660 has been merged, so i close this pr too.

Thanks again for the good work @bgilmer77 👍
